sky/utils/command_runner.py

```diff
@@ -258,6 +258,10 @@ def _rsync(
 
         rsync_command.append(f'-e {shlex.quote(rsh_option)}')
 
+        # Avoid rsync interpreting :, /, and + in node_destination as the
+        # default delimiter for options and arguments.
+        encoded_node_destination = node_destination.replace(':', '%3A').replace(
+            '/', '%2F').replace('+', '%2B')
```

Comment:
Does this also affect the rsync code path for clouds other than k8s?

Reply:
Ahh, that is a good catch! Moved the encoding to kubernetes' specific rsync